apps/mark-scan/frontend/src/pages/contest_screen.tsx

```diff
@@ -23,13 +25,34 @@ function getStartPageUrl() {
 }
 
 export function ContestScreen(): JSX.Element {
+  const [keyEventListenerWasUnbound, setKeyEventListenerWasUnbound] =
+    useState(false);
   const { contests, electionDefinition, precinctId, updateVote, votes } =
     React.useContext(BallotContext);
 
   const isPatDeviceConnected = Boolean(
     api.getIsPatDeviceConnected.useQuery().data
   );
 
+  // handleKeyboardEvent is defined in mark-scan and must be unbound here rather
+  // than in libs/ui
+  const onOpenWriteInKeyboard = useCallback(() => {
+    /* istanbul ignore next - @preserve */
+    if (!isPatDeviceConnected) {
+      document.removeEventListener('keydown', handleKeyboardEvent);
+      setKeyEventListenerWasUnbound(true);
+    }
+  }, [isPatDeviceConnected]);
```

Comment:
Do we have to add the keydown listeners for the write-in keyboard to the document, or could we add them just to the keyboard component? I'm thinking if they worked just on the keyboard component, then we could call `event.stopPropagation` to prevent this document-level listener from triggering, which would remove the need to manage it here.

Reply:
Great suggestion! This unlocks a pretty large simplification

https://github.com/votingworks/vxsuite/pull/6071/commits/a4b1ff3a65160a99718bfa470da3786c8eb474e8

Reply:
Glad to hear it! It took me a while to find this solution so I'm glad it ended up working haha